docs/Gamification_extension_mechanism.rst

```diff
@@ -0,0 +1,175 @@
+.. _Gamification_extension_mechanism:
+
+##################################
+Gamification extension mechanism
+##################################
+
+    The Gamification feature provides a simple extension mechanism that could be used to extend the gamification configuration by adding Rules, Domains and Badges.
+    This mechanism allows also developers to easily add new events without having to make any changes to the Gamification code.
+
+==============
+Add new rules
+==============
+
+To add new rules to the gamification configuration a simple external component plugin should be added to your addon configuration, this plugin should define the following information:
+
+Rule configuration
+~~~~~~~~~~~~~~~~~~
+-  ``rule-title``: The event name of the new rule.
+-  ``rule-description``: The description of the new rule.
+-  ``rule-score``: The score of the new rule (number of gamification points awarded when the event related to the rule is executed).
+-  ``rule-zone``: The Domain name.
+-  ``rule-enable``: Define if the rule is enabled or not
+
+.. code:: xml
+
+    <external-component-plugins>
+        <target-component>org.exoplatform.addons.gamification.service.setting.rule.RuleRegistry</target-component>
+        <component-plugin>
+            <name>pushCode</name>
+            <set-method>addPlugin</set-method>
+            <type>org.exoplatform.addons.gamification.service.setting.rule.model.RuleConfig</type>
+            <init-params>
+                <value-param>
+                    <name>rule-title</name>
+                    <value>pushCode</value>
+                </value-param>
+                <value-param>
+                    <name>rule-description</name>
+                    <value>You pushed a code</value>
+                </value-param>
+                <value-param>
+                    <name>rule-score</name>
+                    <value>15</value>
+                </value-param>
+                <value-param>
+                    <name>rule-zone</name>
+                    <value>Development</value>
+                </value-param>
+                <value-param>
+                    <name>rule-enable</name>
+                    <value>true</value>
+                </value-param>
+            </init-params>
+        </component-plugin>
+    </external-component-plugins>
+
+
+===============
+Add new domains
+===============
+
+To add new domain to the gamification configuration a simple external component plugin should be added to your addon configuration, this plugin should define the following information:
+
+Domain configuration
+~~~~~~~~~~~~~~~~~~~~
+-  ``zone-name``: The name of the new domain.
+-  ``zone-description``: The description of the new domain.
+-  ``zone-priority``: The domain priority.
```

Comment:
It is not clear how priority is used and what are the possible valuea)

Reply:
This property is not used anywhere, but still exist in the code and configuration.